v2/pkg/options/options.go

```diff
@@ -47,9 +48,13 @@ type App struct {
 	AlwaysOnTop       bool
 	// BackgroundColour is the background colour of the window
 	// You can use the options.NewRGB and options.NewRGBA functions to create a new colour
-	BackgroundColour   *RGBA
-	Assets             fs.FS
-	AssetsHandler      http.Handler
+	BackgroundColour *RGBA
+	// Deprecated: Use AssetServer.Assets instead.
+	Assets fs.FS
+	// Deprecated: Use AssetServer.Handler instead.
+	AssetsHandler http.Handler
+	// AssetServer configures the Assets for the application
+	AssetServer        *assetserver.Options
```

Comment:
If we made this a non-pointer, wouldn't the fields be testable for nil anyway? Just wondering if the zero value for this might be a better option.

Reply:
I think that would make it impossible to distinct some cases, e.g. if:

```go
Assets: embedFS,
AssetServer: assetserver.Options{}
```

This would mean `AssetServer.Assets` is nil. Does this now mean the developer wanted to have `nil` assets but just forgot removing it from the deprecated `Assets` or did they want the `Assets` and forgot to add them to the `AssetServer` struct.

For the check of inconsistency it's much easier to have struct-pointer. If it's nil one can be assured the developer didn't want to change anything and was using the deprecated options. If it has been set, the developer wanted to migrate to the `AssetServer` option, but forgot to remove the old assignments.

Reply:
But yeah, if migration is over I would also opt for having a struct instead of a struct pointer there.